### DISCUSSION TITLE
Real time projected texture preview in game | tool made in vscript

### DISCUSSION THREAD

**Discussion by TheEnderek0** [StrataSource/Portal-2-Community-Edition#1050] [Resources]
Script made by me to place and test projected textures in game in real time.

Using this you can open a map, load the script and place the env_projectedtextures in game directly, then transform them to hammer.
To place them in hammer you can either modify the vmf file directly, or place the texture at 0 0 0, then using the transform tool transform it's position and angles.

Report any issues to me on discord or here.

Download: [here](https://cdn.discordapp.com/attachments/793465587409485856/1025498952730157097/ptex_test.nut)

**Projected Texture Tester by Enderek0**

Instructions on usage:

**Setup:**

Firstly you need to place this script in `scripts/vscripts/` in a mounted filesystem, so for example `Portal 2 Community Edition/p2ce/scripts/vscripts/`

Launch the game:
1. Open the map you want to test on.
2. Open the console, in order to use this tool you will need to use the console unfortunatelly.
3. You need to enable developer mode, for that type `developer 1`.
    **-> If the console tells you the max allowed developer is 0, type `sv_max_allowed_developer 2`.**
4. Initialise the script, type: `script_execute ptex_test`.

**Pro tip: use the command `clear` to clear the console.**

Usage:
After you setup you will fire commands of specific functions to operate, these are like so:
`script x`, where x is the function you want to fire, so like `script NewEntity()`, `script Fov(80)`.

Functions:
These are the functions that change the properties of the projected texture as well as spawn it, disable etc.

**- Spawn/Stop:**
>    `NewEntity()` - Spawns a new projected texture with default keyvalues. Automatically disables the previous one if it was selected.
>    `PrePlacedEntity(string)` - Selects a projected texture with a name, instead of name type the actual name of the texture. Automatically disables the previous one if it was selected.
>     `Disable()` - Stops the preview of current texture, prints all information to the console.
>     `Freeze()` - Pauses/resumes the movement of the current texture.

**- Keyvalues: (All of the number parameters here support fractions.)**
>    `Fov(float)` - Sets the field of view. | e.g. Fov(90)
>    `NearZ(float)` - NearZ.
>    `FarZ(Float)` - FarZ.
>
>    `Color(string)` - Sets the color, the parameter should be typed in like so: Color("255 255 255 200")
>    `ColorTransitionTime(float)` - Sets the transition time between colors.
>    `Brightness(float)` - Sets the brightness scale.
>
>    `Texture(string)` - Sets the texture name. | e.g. Texture("effects/flashlight001")
>    `Frame(integer)` - Sets the frame of the texture.
>
>    `Name(string)` - Sets the targetname.
    Textures spawned by the script automatically get called PTEX* where * is a next number relative to the last one, so PTEX0, PTEX1, PTEX2.
    You can change the name yourself. Use `ent_text env_projectedtexture` in the console to view all of the entities.
    Here's a neat thing: You can select an entity that you created with this script by using PrePlacedEntity() !
    Then you can modify it more or dump the information again by using Disable().
    That's mainly the reason this whole Name scripting was written.

Credit not required.

**Comment by JJL772** [comment]
Is this compatible with Portal 2?

**Reply by TheEnderek0** [reply]
No

**Comment by kiddu25** [comment]
hey ! can you reupload this pls ?

**Reply by TheEnderek0** [reply]
Copy the link and paste it into discord anywhere to download it.
Discord doesn't allow accessing the download links from 3rd party apps now.

**Reply by kiddu25** [reply]
> Copy the link and paste it into discord anywhere to download it. Discord doesn't allow accessing the download links from 3rd party apps now.

thanks !

**Reply by HugoBDesigner** [reply]
> Copy the link and paste it into discord anywhere to download it. Discord doesn't allow accessing the download links from 3rd party apps now.

Consider uploading the script to this page directly. One of the goals of this repository is to decouple resources from being Discord-exclusively, especially for people who don't have accounts, are logged off, or in the (unlikely but still possible) case of Discord dying or changing domains or other such :P

**Reply by TheEnderek0** [reply]
I think I've tried to, but github has weird restrictions as to what filetypes you can upload onto the service.
I've been considering making this an actual workshop item instead, when I'll have time I'll definitely make it one.

**Reply by craftablescience** [reply]
I recommend adding the file to a zip and uploading that if GitHub blocks it.
